pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/ByteArrayBufferedData.java

```diff
@@ -84,10 +82,10 @@ public byte getByte(final long offset) {
     public long getBytes(final long offset, @NonNull final byte[] dst, final int dstOffset, final int maxLength) {
         validateLen(maxLength);
         final long len = Math.min(maxLength, length() - offset);
```

Comment:
`offset` is not validated at this point yet, it can be (for example) negative. Now assume the following call: `getBytes(-1, dst, 0, length + 1)`. It would initialize `len` to `length() + 1`, which doesn't look good

Reply:
Added a separate check.